### PR TITLE
[MM-48544] Refactor calls recorder

### PR DESCRIPTION
**Pull request by streamer45**
#### Summary

PR refactors the existing calls recorder logic by making it a little bit more configurable and also updates it to use the new, plugin exposed API for uploading recordings.

#### Related PRs

https://github.com/mattermost/mattermost-plugin-calls/pull/246

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-48544

### CODE REVIEW

File: cmd/recorder/config.go [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+	}
+	if cfg.SiteURL == "" {
+		return fmt.Errorf("SiteURL cannot be empty")
+	}
+	if cfg.CallID == "" {
+		return fmt.Errorf("CallID cannot be empty")
+	}
+	if cfg.ThreadID == "" {
+		return fmt.Errorf("ThreadID cannot be empty")
+	}
```
Comment (streamer45) [inline cmd/recorder/config.go:65]: I am not that all in love with having to pass this but until we start persisting the call state/info we need to keep some state here as the call could be over when the processing completes on this side.

File: .gitignore [hunk 2]
```diff
@@ -2,3 +2,4 @@
 dist/
 
 .config.env*
+*.vim
```
Comment (cpoile) [inline .gitignore:5]: I would also like to ignore vim

File: cmd/recorder/config.go (hunk 1 above)
Comment (cpoile) [reply cmd/recorder/config.go:65]: I don't see why it would be a problem.

**Review by cpoile — APPROVED** [review]
Looks great, much cleaner!

File: cmd/recorder/config.go (hunk 1 above)
Comment (streamer45) [reply cmd/recorder/config.go:65]: Not a problem per se but it feels like exposing implementation details.

File: .gitignore (hunk 2 above)
Comment (streamer45) [reply .gitignore:5]: Don't get me started :speak_no_evil:

**Pull request merged**